build.rs

```diff
@@ -75,16 +97,31 @@ fn build(metadata: &Metadata) {
         ];
 
         // If we're cross-compiling, let configure know.
-        if metadata.host != metadata.target {
+        if metadata.is_cross_compiling() {
+            configure_args.push(format!("--build={}", metadata.host));
+            // `--host` is the target platform in autotools parlance, not the
+            // host platform.
             configure_args.push(format!("--host={}", metadata.target));
         }
 
         let configure_path = Path::new("krb5").join("src").join("configure");
-        cmd(configure_path, &configure_args)
+
+        let mut configure = cmd(configure_path, &configure_args)
             .dir(&metadata.build_dir)
-            .env_remove("CONFIG_SITE")
-            .run()
-            .expect("configure failed");
+            .env_remove("CONFIG_SITE");
+        if let Some(cc) = metadata.cc() {
+            configure = configure.env("CC", cc)
+        }
+        if let Some(ar) = metadata.ar() {
+            configure = configure.env("AR", ar)
+        }
+        if metadata.is_cross_compiling() {
+            configure = configure
+                .env("krb5_cv_attr_constructor_destructor", "yes")
+                .env("ac_cv_func_regcomp", "yes")
+                .env("ac_cv_printf_positional", "yes")
```

Comment:
We've historical held the position that it is inappropriate to set these flags in this crate, as we don't actually know that the target system supports these features that configure is checking—they're in theory something that whoever's doing the cross compile needs to set for themselves.

That said, it does seem like the answer is `yes` for basically all the cross-compiling targets that anyone cares about these days...